winpgntc.c

```diff
@@ -107,14 +107,18 @@ agent_query(void *buf)
             UnmapViewOfFile(p);
             CloseHandle(filemap);
             LocalFree(psd);
+            psd = NULL;
             free(usersid);
-
+            usersid = NULL;
             if (id > 0)
                 return;
         }
 
-        LocalFree(psd);
-        free(usersid);
+        if (psd)
+            LocalFree(psd);
+        if (usersid)
+            free(usersid);
```

Comment:
`LocalFree` and `free` are fine with `NULL`, so these checks aren't really necessary.

Reply:
I've updated the code.